woocommerce/woocommerce-ios
# Edit Product Inventory: new option to check if SKU already exists

Fixes #1710 

## Description
This implementation improve how we check if a product SKU already exists on the website while the user is editing it.
The solution was to implement this new endpoint `/wc/v3/products/&_method=get&sku={sku}` and use also the parameter `_fields` to obtain only the `sku` field (this will increase the performance of the API speed).
Another solution was to call the API every time the user is typing the SKU, implementing a throttling method to make sure we call the API not much frequently.
In the end, I catch the SKU text field to be the first responder. Initially, when the error appears/disappears during the table view refresh, the keyboard will disappear.

## Changes
- Implemented the `ProductSkuMapper` + tests
- Implement the new method `searchSku` in `ProductsRemote` + tests 
- Implemented a new tool class `Throttler` used for throttling a function
- Implemented the new SKU handling logic in `ProductInventorySettingsViewController`
- Implemented the new `searchSku()` method in `ProductStore` + tests

## Testing
1) Navigate to a product detail
2) Tap edit
3) Tap inventory
4) Try to type a new SKU. You will not see the SKU error and you are able to save it.
5) Try to type an existing SKU. You will be able to see the SKU error
6) Try to delete the SKU and retype the old SKU. You will not see the SKU error.

Update release notes:
